Remove python boolean check from adjust_gamma

Timen opened the pull request:
The adjust_gamma operation used a python boolean to check if gamma was positive.  This however removes any possibility of using a scalar tensor, which means it can't be used to apply random augmentations during training.

It is fixed by using a tf.assert_positive statement.

Timen committed:
Update image_ops_impl.py

tensorflow-jenkins commented:
Can one of the admins verify this patch?

Timen commented:
signed

Timen committed:
Update image_ops_impl.py

martinwicke commented:
@drpngx could you look at this?

drpngx commented:
Any reason not to use `tf.assert_non_negative` instead? That allows `gamma=0`, as in the original code.

drpngx commented:
Also, if `gamma` can be a vector, we'd need to change the doc, and add a test.

martinwicke commented:
@drpngx could you look at this?

@Timen It's often still useful to have a static check -- can you check whether gamma is a tensor (with `tf.framework.is_tensor`) and only if it is, add the assert? That way you get the static check if it is available.

martinwicke commented:
(sorry @drpngx, I started that comment before you replied, please ignore)

Timen committed:
Update image_ops_impl.py

Added tensor not tensor based switch, to determine whether or not to use the tensor based assert or python boolean assert.

Timen commented:
Added requested if else based on whether or not gamma is a tensor.

Timen commented:
Not sure why the tests fail I can't see any of the failure reports.

drpngx committed:
Minor fixes

Check that gamma is a tensor, not image.
Update comment per style and correctness.
Print out gamma value in static error check.

drpngx commented:
Two failures:
```
FAIL: Found 2 non-whitelited pylint errors:
tensorflow/python/ops/image_ops_impl.py:1011: [W0311(bad-indentation), ] Bad indentation. Found 7 spaces, expected 6

tensorflow/python/ops/image_ops_impl.py:1012: [W0311(bad-indentation), ] Bad indentation. Found 10 spaces, expected 8
```

and

```
======================================================================
ERROR: test_adjust_gamma_greater_one (__main__.AdjustGamma)
Verifying the output with expected results for gamma
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/botexec/bazel-out/k8-opt/bin/tensorflow/python/image_ops_test.runfiles/org_tensorflow/tensorflow/python/ops/image_ops_test.py", line 235, in test_adjust_gamma_greater_one
    y = image_ops.adjust_gamma(x_np, gamma=2)
  File "/tmp/botexec/bazel-out/k8-opt/bin/tensorflow/python/image_ops_test.runfiles/org_tensorflow/tensorflow/python/ops/image_ops_impl.py", line 1008, in adjust_gamma
    if tf.framework.is_tensor(gamma):
NameError: global name 'tf' is not defined

----------------------------------------------------------------------
Ran 28 tests in 1.458s

FAILED (errors=1)
```

You can also run the test locally!
```
bazel test -c opt //tensorflow/python:image_ops_test
```

Commit:
Fixed bugs in gamma implementation, switch to using _assert for tensor and python float assertion. Added unit tests.

Timen commented:
Implemented a new assertion structure using the available _assert method, added unit tests for gamma < 0 in python float and tensor form. Local tests passed successfully.

drpngx commented:
Jenkins, test this please.

drpngx committed:
Use TF dtype for TF constant

drpngx committed:
Fix style

drpngx committed:
Fix typo in previous commit

drpngx commented:
Jenkins, test this please.

Timen committed:
Update image_ops_test.py

Timen commented:
fixed tensorflow dtype prefix

Timen commented:
Local tests passed.

drpngx commented:
Jenkins, test this please/

drpngx commented:
Jenkins, test this please.
